OpenRA.Mods.Common/Activities/KillSelf.cs

```diff
@@ -0,0 +1,36 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Activities;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public class KillSelf : Activity
```

Comment:
This is a small single-purpose activity that is only really useful for `AttackSuicides`, and will become even more specialized in a followup PR that fixes #16061.  Can you please move this definition inside the trait, similarly to `AttackFollow`'s `AttackActivity`?

https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs#L64

Reply:
Please also squash all the `AttackSuicides` changes into one commit

Reply:
Done and done.